Kernel/Tasks/Process.cpp

```diff
@@ -293,16 +293,33 @@ void Process::unprotect_data()
     });
 }
 
-ErrorOr<Process::ProcessAndFirstThread> Process::create_with_forked_name(UserID uid, GroupID gid, ProcessID ppid, bool is_kernel_process, NonnullRefPtr<VFSRootContext> vfs_root_context, NonnullRefPtr<HostnameContext> hostname_context, RefPtr<Custody> current_directory, RefPtr<Custody> executable, RefPtr<TTY> tty, Process* fork_parent)
+ErrorOr<Process::ProcessAndFirstThread> Process::create_from_fork(Process& parent)
 {
+    VERIFY(!parent.is_kernel_process());
+    auto const& credentials = parent.credentials();
+
     Process::Name name {};
-    Process::current().name().with([&name](auto& process_name) {
+    parent.name().with([&name](auto& process_name) {
         name.store_characters(process_name.representable_view());
     });
-    return TRY(Process::create(name.representable_view(), uid, gid, ppid, is_kernel_process, move(vfs_root_context), move(hostname_context), current_directory, executable, tty, fork_parent));
+
+    return TRY(Process::create_impl(name.representable_view(),
+        credentials->uid(), credentials->gid(), parent.pid(),
+        parent.is_kernel_process(), parent.vfs_root_context(),
+        parent.hostname_context(), parent.current_directory(),
+        parent.executable(), parent.tty(), &parent));
+}
+
+ErrorOr<Process::ProcessAndFirstThread> Process::create_spawned(UserID uid, GroupID gid, ProcessID ppid, NonnullRefPtr<VFSRootContext> vfs_root_context, NonnullRefPtr<HostnameContext> hostname_context, NonnullRefPtr<Custody> current_directory, RefPtr<TTY> tty)
+{
+    return TRY(Process::create_impl("Spawned Process"sv,
```

Comment:
This name should never be visible (as explained in one of the other comments).

So I think we should maybe make it clearer that this is just a placeholder name for a process that has no executable yet?